### ISSUE TITLE
Difficulty levels

### ISSUE THREAD

**Issue by Sebeb** [dhubber/mazed-up#11]
Since the size of each level, as well as the number of enemies, traps, and pills, are all set in the level generation algorithm, it should be quite easy to add difficulty levels to the game. Probably just a basic easy, normal, hard selection. This would allow players put off by the difficulty to start easier, or the players who have already completed normal to try something more challenging.

**Comment by dhubber** [comment]
Yes, I was thinking two ideas for this actually.  First, what you suggest so perhaps the maze sizes will be the same but changing the difficulty will change 
1) the required number of pills to complete the level
2) the density/number of traps enemies

The second idea is to use a different maze generating algorithm.  There are alternative ones we could use (I had in the back of my mind during the jam but never implemented more than 1) which can either make the levels less/more challenging and/or just give them a different feel so can change slightly the nature of the mazes.  Any thoughts on this?

**Comment by Sebeb** [comment]
Actually, I was thinking of potentially scrapping this idea, in favor for widening the gap between pills required in a level and number of pills spawned in a level. That way, if the player wants to complete the game in 'easy' mode, they can just go through the game picking up the bare minimum of pills. And if they want to complete it in hard mode, they go through and attempt to pick up every pill. And of course, how they play would be reflected in their score. Maybe this could be enough? in terms of testing and balancing, it would certainly be easier to test and balance one version of the game.

Regarding the maze generation algorithm, I think it could be a good idea to add some more interest and features to it anyway. While what we have is very functional and simple, if you wanted to spend some time looking at how to make mazes slightly more dynamic, I think it could really add something to the game!
